subsys/mpsl/pin_debug/mpsl_pin_debug_radio_core.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <nrfx_gpiote.h>
+#include <helpers/nrfx_gppi.h>
+#include <hal/nrf_radio.h>
+
+#ifdef PPI_PRESENT
+#include <nrfx_ppi.h>
+#elif defined(DPPI_PRESENT)
+#include <mpsl_dppi_protocol_api.h>
+#endif
+
+LOG_MODULE_REGISTER(mpsl_radio_pin_debug, CONFIG_MPSL_LOG_LEVEL);
+
+static int m_ppi_config(void)
+{
+#if defined(PPI_PRESENT)
+	uint8_t ppi_chan_radio_ready;
+	uint8_t ppi_chan_radio_address;
+	uint8_t ppi_chan_radio_end;
+	uint8_t ppi_chan_radio_disabled;
+
+	if (nrfx_ppi_channel_alloc(&ppi_chan_radio_ready) != NRFX_SUCCESS) {
+		LOG_ERR("Failed allocating PPI chan");
+		return -1;
```

Comment:
You can consider using -NRF_EPERM or other relevant error codes from include/nrf_errno.h